### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by lrhn**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,80 @@
 [![pub package](https://img.shields.io/pub/v/characters.svg)](https://pub.dev/packages/characters)
 [![Build Status](https://travis-ci.org/dart-lang/characters.svg?branch=master)](https://travis-ci.org/dart-lang/characters)
 
-**NOTE**: This package is in technical preview, and breaking API changes are to be expected.
+ ***\*NOTE\****: This package is in technical preview, and breaking API changes are to be expected.
```
Comment (mit-mit) [inline README.md]: Remove this again

File: README.md [hunk 2]
```diff
@@ -1,10 +1,80 @@
...
 
-`Characters` are strings viewed as sequences of *user-perceived character*s,
-also know as [Unicode (extended) grapheme clusters](https://unicode.org/reports/tr29/#Grapheme_Cluster_Boundaries).
+[Characters][] are strings viewed as sequences of **user-perceived character**s, also know as [Unicode (extended) grapheme clusters][Grapheme Clusters].
```
Comment (mit-mit) [inline README.md]: Why are the the paragraphs with a leading single space?

File: README.md [hunk 3]
```diff
@@ -1,10 +1,80 @@
...
+
+The `Characters` class allows access to the individual characters of a string, and a way to navigate back and forth between them using a `CharacterRange`.
+
+## Unicode Characters and Representations
```
Comment (mit-mit) [inline README.md]: We use sentence casing for headings in Dart documentation

File: README.md [hunk 4]
```diff
@@ -1,10 +1,80 @@
...
 [![Build Status](https://travis-ci.org/dart-lang/characters.svg?branch=master)](https://travis-ci.org/dart-lang/characters)
 
-**NOTE**: This package is in technical preview, and breaking API changes are to be expected.
+ ***\*NOTE\****: This package is in technical preview, and breaking API changes are to be expected.
 
-`Characters` are strings viewed as sequences of *user-perceived character*s,
-also know as [Unicode (extended) grapheme clusters](https://unicode.org/reports/tr29/#Grapheme_Cluster_Boundaries).
+[Characters][] are strings viewed as sequences of **user-perceived character**s, also know as [Unicode (extended) grapheme clusters][Grapheme Clusters].
+
+The `Characters` class allows access to the individual characters of a string, and a way to navigate back and forth between them using a `CharacterRange`.
```
Comment (mit-mit) [inline README.md]: Wrap lines to 80 characters?

File: README.md [hunk 5]
```diff
@@ -1,10 +1,80 @@
...
+| [Code points][]       | [Runes][]                                                    | The Unicode unit of meaning.                                 | `string.runes`                                               |
+| [Grapheme Clusters][] | [Characters][]                                               | Human perceived character. One or more code points.          | `string.characters`                                          |
+| [Glyphs][]            |                                                              | Visual rendering of grapheme clusters.                       | `print(string)`                                              |
+
+A Dart `String` is a sequence of UTF-16 code units, just like strings in JavaScript and Java. The runtime system decides on the underlying physical representation.
+
+That makes plain strings inadequate when needing to manipulate the text that a user is viewing, or entering, because they are working at the grapheme cluster level.
```
Comment (mit-mit) [inline README.md]: Is "they" here the user? Then maybe "because humans perceive characters at the grapmeme cluster level"?

File: README.md [hunk 6]
```diff
@@ -1,10 +1,80 @@
...
+
+For example, to show an abbreviated text like "A long text tha&mldr;", it's necessary to find the first 15 *glyphs*, not just the first fifteen code units or code points. For non-ASCII texts, that's not always the same.
```
Comment (mit-mit) [inline README.md]: Maybe use an example string that actually has a non-ASCII character in it?

Comment (mit-mit) [reply README.md]: E.g. `A long text 📄 t...`

File: README.md [hunk 7]
```diff
@@ -1,10 +1,80 @@
...
+
+Whenever you need to manipulate strings at the character level, you should be using the `Characters` type, not the methods of the string class.
```
Comment (mit-mit) [inline README.md]: s/string class/String class/ ?

File: README.md [hunk 8]
```diff
@@ -1,10 +1,80 @@
...
+
+## The Characters Class
```
Comment (mit-mit) [inline README.md]: s/Class/class/

File: README.md (hunk 2 above)
Comment (kwalrath) [reply README.md]: know as -> known as

**Comment by kwalrath** [comment]
I happened to take a quick look. @mit-mit's copyedit-level suggestions are exactly what I would've said. 🎉

File: README.md (hunk 2 above)
Comment (lrhn) [reply README.md]: @mit-mit Probably a side-effect of cut-n-paste'ing from github to my markdown editor. Will fix.

File: README.md (hunk 1 above)
Comment (lrhn) [reply README.md]: Change not intended, will revert.

File: README.md (hunk 3 above)
Comment (lrhn) [reply README.md]: ACK. Will adapt.

File: README.md (hunk 4 above)
Comment (lrhn) [reply README.md]: How ... conventional! Will do.

File: README.md (hunk 5 above)
Comment (lrhn) [reply README.md]: "They" actually refers to "plain strings". Will rewrite to make unambiguous.

File: README.md (hunk 7 above)
Comment (lrhn) [reply README.md]: Either "string class" or "`String` class" - if it's a source name, it's code quoted, if it's just a general concept, it's not. 
Will change to the latter.

File: README.md (hunk 8 above)
Comment (lrhn) [reply README.md]: ack

**Comment by kwalrath** [comment]
You used semantic line breaks, my favorite!

**Comment by mit-mit** [comment]
@lrhn can we merge this? Then I'll publish the 1.0 version.

**Comment by lrhn** [comment]
Done.

**Pull request merged**